.github/workflows/check.yml

```diff
@@ -32,44 +31,84 @@ jobs:
         - 3.8
         - 3.7
         - 3.6
+        - 3.5
+        - pypy3
         - 2.7
         - pypy2
-        - pypy3
-
+        include:
+        - {os: MacOs, py: brew@py3}
+        - {os: MacOs, py: brew@py2}
     steps:
-    - uses: actions/checkout@v2
-    - uses: tox-dev/action@v1.0.0
-      with:
-        python-version: ${{ matrix.py }}
-
+    - name: install OS dependencies
+      run:   |
+        ${{ runner.os == 'Linux' && 'sudo apt-get update -y && sudo apt-get install fish csh' || true}}
+        ${{ runner.os == 'macOS' && 'brew update -vvv && brew install fish tcsh' || true}}
+        ${{ runner.os == 'Windows' && ( matrix.py == '2.7' || matrix.py == 'pypy2' ) && 'choco install vcpython27 --yes -f' || true }}
+      shell: bash
     - name: setup python for tox
       uses: actions/setup-python@v2
       with:
         python-version: 3.8
     - name: install tox
       run: python -m pip install tox
+    - uses: actions/checkout@v2
+    - name: use local virtualenv for tox
+      run: python -m pip install ${{ runner.os == 'Windows' && matrix.py == 'pypy3' && 'virtualenv<20' || '.'}}
```

Comment:
FYI you can put logic into separate steps and pass values via outputs: https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-output-parameter

Reply:
I think in this case this is the best way, but thanks 👍